ibm_mq/datadog_checks/ibm_mq/queue_metric_collector.py

```diff
@@ -0,0 +1,180 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+from six import iteritems
+
+from datadog_checks.base import AgentCheck, ensure_bytes, ensure_unicode
+from datadog_checks.ibm_mq.metrics import GAUGE
+
+from . import metrics
+
+try:
+    import pymqi
+except ImportError as e:
+    pymqiException = e
+    pymqi = None
+else:
+    # Since pymqi is not be available/installed on win/macOS when running e2e,
+    # we load the following constants only pymqi import succeed
+    SUPPORTED_QUEUE_TYPES = [pymqi.CMQC.MQQT_LOCAL, pymqi.CMQC.MQQT_MODEL]
+
+
+class QueueMetricCollector:
+    QUEUE_SERVICE_CHECK = 'ibm_mq.queue'
+    QUEUE_MANAGER_SERVICE_CHECK = 'ibm_mq.queue_manager'
+
+    def __init__(self, config, service_check, warning, send_metric, log):
+        self.config = config
+        self.service_check = service_check
+        self.warning = warning
+        self.send_metric = send_metric
+        self.log = log
```

Comment:
Doesn't feel right to pass all those methods of `AgentCheck` but don't have better idea without heavier refactoring.

Same comment for the other collector.

We can probably improve that separately.

Reply:
yes, we could make these classes inherit from agent check.

Reply:
We do have an example of passing self as check in the QueryManager. I felt more confortable passing only the required methods though